kafka/collector.go

```diff
@@ -0,0 +1,133 @@
+package kafka
+
+import (
+	"errors"
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strconv"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+	exporter "github.com/kawamuray/prometheus-kafka-consumer-group-exporter"
+)
+
+// Fields separator is different between kafka versions
+// 0.9.X  => ", "
+// 0.10.X => \s+
+var consumerGroupCommandDescribeOutputSeparatorRegexp = regexp.MustCompile(`,?\s+`)
+
+// ConsumerGroupClient queries Kafka for consumer groups and their current log
+// and offset states.
+type ConsumerGroupClient struct {
+	BootstrapServer         string
+	ConsumerGroupCommandPath string
+}
+
+func (col *ConsumerGroupClient) execConsumerGroupCommand(args ...string) (string, error) {
+	allArgs := append([]string{"--new-consumer", "--bootstrap-server", col.BootstrapServer}, args...)
+	cmd := exec.Command(col.ConsumerGroupCommandPath, allArgs...)
+	output, err := cmd.CombinedOutput()
```

Comment:
We've seen `--list` script execution lock-up forever. No sure why. Before merging this pull request, we should probably add a timeout here that kills that script after a while.